crates/cargo-bsp/src/cargo_communication/check/cargo_message_to_package_info.rs

```diff
@@ -0,0 +1,136 @@
+//! Maps information from Cargo Messages produces by `cargo check` command to
+//! RustPackage information.
+
+use crate::utils::uri::file_uri;
+use bsp_types::extensions::RustCfgOptions;
+use bsp_types::Uri;
+use cargo_metadata::{Artifact, BuildScript, Package};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+const DYNAMIC_LIBRARY_EXTENSIONS: [&str; 3] = ["dll", "so", "dylib"];
```

Comment:
to nie brzmi specjalnie future-proof

Reply:
To jest przekopiowane na podstawie tego jak te dane oblicza sobie normalnie intellij-rust i właśnie w taki sposób to robi